### PR TITLE
Pagination for REST clients

### LINKED ISSUE #2511: Paging with StorageClient
Is it possible to page with the `StorageClient` in the current version of the .NET Api?

I've looked at the source and can see the nextPageToken is being set by an internal `PageManager`, for one big page, but there doesn't seem to be a manual way of:

1. Fetch 100 items, page 1
2. Fetch 200 items, page 2
3. so on..

### PR DESCRIPTION
**Pull request by jskeet**
Fixes #2511.

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/ListDatasetsOptions.cs [hunk 1]
```diff
@@ -41,6 +41,13 @@ public sealed class ListDatasetsOptions
         /// </summary>
         public bool? IncludeHidden { get; set; }
 
+        /// <summary>
+        /// If set, this is the token is used to indicate a continued list operation.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/ListDatasetsOptions.cs]: nit:
this is the token that is used 
otr
this token is used

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/ListJobsOptions.cs [hunk 2]
```diff
@@ -44,6 +45,13 @@ public sealed class ListJobsOptions
         /// </summary>
         public ProjectionEnum? Projection { get; set; }
 
+        /// <summary>
+        /// If set, this is the token is used to indicate a continued list operation.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/ListJobsOptions.cs]: nit: same as above

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/ListProjectsOptions.cs [hunk 3]
```diff
@@ -27,12 +28,24 @@ public sealed class ListProjectsOptions
         /// </summary>
         public int? PageSize { get; set; }
 
+        /// <summary>
+        /// If set, this is the token is used to indicate a continued list operation.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/ListProjectsOptions.cs]: nit: same

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/ListTablesOptions.cs [hunk 4]
```diff
@@ -27,12 +28,23 @@ public sealed class ListTablesOptions
         /// </summary>
         public int? PageSize { get; set; }
 
+        /// <summary>
+        /// If set, this is the token is used to indicate a continued list operation.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/ListTablesOptions.cs]: nit: same

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/ListBucketsOptions.cs [hunk 5]
```diff
@@ -39,6 +39,13 @@ public sealed class ListBucketsOptions
         /// </summary>
         public Projection? Projection { get; set; }
 
+        /// <summary>
+        /// If set, this is the token is used to indicate a continued list operation.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/ListBucketsOptions.cs]: nit: same as above

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/ListObjectsOptions.cs [hunk 6]
```diff
@@ -57,6 +57,13 @@ public sealed class ListObjectsOptions
         /// </summary>
         public string UserProject { get; set; }
 
+        /// <summary>
+        /// If set, this is the token is used to indicate a continued list operation.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/ListObjectsOptions.cs]: nit: same

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/ListDatasetsOptions.cs (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/ListDatasetsOptions.cs]: Thanks. Will change to "this token is used" throughout.

**Comment by jskeet** [comment]
Will merge on green, then send another PR for a new version.

**Pull request merged**